import-service/handler.js

```diff
@@ -0,0 +1,100 @@
+const AWS = require('aws-sdk');
+const csv = require('csv-parser');
+
+const BUCKET = 'product-service.imported-products';
+
+module.exports.importProductsFile = async function(event) {
+  try {
+    const s3 = new AWS.S3({ region: 'eu-west-1' });
+    const filename = event['queryStringParameters']['name'] || 'products';
+    const productsFilePath = `uploaded/${filename}.csv`;
+
+    const params = {
+      Bucket: BUCKET,
+      Key: productsFilePath,
+      Expires: 60,
+      ContentType: 'text/csv',
+    };
+
+    const uploadURL = await new Promise((resolve, reject) => {
+      s3.getSignedUrl('putObject', params, function (err, url) {
+        if (err) {
+          reject(err)
+        }
+        resolve(url)
+      })
+    })
+
+    return {
+      statusCode: 200,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Credentials': true,
+      },
+      body: JSON.stringify({ uploadURL }),
+    }
+  } catch (e) {
+    return {
+      statusCode: 500,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Credentials': true,
+      },
+      body: JSON.stringify({ error: e }),
```

Comment:
Are you sure you want to return actual error to the user? You cannot control it and it might contain some sensitive information. I would suggest to return simply 'Server error'